tidy3d/components/data/near2far_radiation_vectors.py

```diff
@@ -68,11 +73,11 @@ class Near2Far(Tidy3dBaseModel):
         description="Tuple of each :class:`.Near2FarSurface` to use as source of near field.",
     )
 
-    frequency: float = pydantic.Field(
-        ...,
-        title="Frequency",
-        description="Frequency to select from each :class:`.Near2FarSurface` for projection.",
-        units=HERTZ,
+    resample: bool = pydantic.Field(
```

Comment:
That's a good idea - I made `pts_per_wavelength` 10 by default to mimic the current behaviour, but made it so that it can be set to `None` to turn off resampling (also made this clear in the doc strings). The `resample` field is then no longer needed, which makes things simpler.